-1

Tested at: 4ba24111ea6743ecb9d3ca7e98fa87d8ce58b3a9

You can see the results of the tests here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-21746/25184/summary.html

I found follow errors while testing this PR

**Failed tests:  Build**

* **Build**:

I found an error when building: 
 
 <pre>Entering library rule at src/CalibTracker/SiStripESProducers/plugins
>> Compiling  /build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_10_0_X_2017-12-19-1100/src/CalibTracker/SiStripESProducers/src/SiStripFedCablingESProducer.cc 
>> Compiling  /build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_10_0_X_2017-12-19-1100/src/CalibTracker/SiStripESProducers/src/SiStripHashedDetIdESProducer.cc 
>> Compiling  /build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_10_0_X_2017-12-19-1100/src/CalibTracker/SiStripESProducers/src/SiStripGainESSource.cc 
/build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_10_0_X_2017-12-19-1100/src/HLTrigger/HLTcore/src/TriggerExpressionData.cc: In member function 'bool triggerExpression::Data::setEvent(const edm::Event&, const edm::EventSetup&)':
/build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_10_0_X_2017-12-19-1100/src/HLTrigger/HLTcore/src/TriggerExpressionData.cc:48:70: error: call of overloaded 'get(const edm::Event&, edm::EDGetTokenT<BXVector<GlobalAlgBlk> >&)' is ambiguous
     auto l1t = get<GlobalAlgBlkBxCollection>(event, m_l1tResultsToken);
                                                                      ^
/build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_10_0_X_2017-12-19-1100/src/HLTrigger/HLTcore/src/TriggerExpressionData.cc:20:11: note: candidate: const T* triggerExpression::get(const edm::Event&, const edm::EDGetTokenT<T>&) [with T = BXVector<GlobalAlgBlk>]
 const T * get(const edm::Event & event, const edm::EDGetTokenT<T> & token) {
           ^~~
</pre>

--- a/DQM/HLTEvF/plugins/TriggerBxMonitor.cc
+++ b/DQM/HLTEvF/plugins/TriggerBxMonitor.cc
@@ -33,26 +33,6 @@
 #include "DQMServices/Core/interface/DQMEDAnalyzer.h"
 #include "DQMServices/Core/interface/MonitorElement.h"
 
-// helper functions
-template <typename T>
-static
-const T & get(const edm::Event & event, const edm::EDGetTokenT<T> & token) {
-  edm::Handle<T> handle;
-  event.getByToken(token, handle);
-  if (not handle.isValid())
-    throw * handle.whyFailed();
-  return * handle.product();
-}
-
-template <typename R, typename T>
-static
-const T & get(const edm::EventSetup & setup) {
-  edm::ESHandle<T> handle;
-  setup.get<R>().get(handle);
-  return * handle.product();
-}
-
-
 class TriggerBxMonitor : public DQMEDAnalyzer {
 public:
   explicit TriggerBxMonitor(edm::ParameterSet const &);
@@ -171,7 +151,7 @@ void TriggerBxMonitor::dqmBeginRun(edm::Run const & run, edm::EventSetup const &
   }
 
   // cache the L1 trigger menu
-  m_l1tMenu = & get<L1TUtmTriggerMenuRcd, L1TUtmTriggerMenu>(setup);
+  m_l1tMenu = & edm::get<L1TUtmTriggerMenu, L1TUtmTriggerMenuRcd>(setup);
   if (m_l1tMenu) {
     if (m_make_1d_plots) {
       m_l1t_bx.clear();
@@ -295,7 +275,7 @@ void TriggerBxMonitor::analyze(edm::Event const & event, edm::EventSetup const &
 
   // monitor the bx distribution for the L1 triggers
   if (m_l1tMenu) {
-    auto const & bxvector = get<GlobalAlgBlkBxCollection>(event, m_l1t_results);
+    auto const & bxvector = edm::get(event, m_l1t_results);
     if (not bxvector.isEmpty(0)) {
       auto const & results = bxvector.at(0, 0);
       for (unsigned int i = 0; i < GlobalAlgBlk::maxPhysicsTriggers; ++i)
@@ -311,7 +291,7 @@ void TriggerBxMonitor::analyze(edm::Event const & event, edm::EventSetup const &
 
   // monitor the bx distribution for the HLT triggers
   if (m_hltConfig.inited()) {
-    auto const & hltResults = get<edm::TriggerResults>(event, m_hlt_results);
+    auto const & hltResults = edm::get(event, m_hlt_results);
     for (unsigned int i = 0; i < hltResults.size(); ++i) {
       if (hltResults.at(i).accept()) {
         if (m_make_1d_plots and m_hlt_bx.at(i))

--- a/DQM/HLTEvF/plugins/TriggerBxVsOrbitMonitor.cc
+++ b/DQM/HLTEvF/plugins/TriggerBxVsOrbitMonitor.cc
@@ -2,11 +2,6 @@
 #include <string>
 #include <cstring>
 
-// boost headers
-#include <boost/regex.hpp>
-#include <boost/format.hpp>
-#include <boost/lexical_cast.hpp>
-
 // Root headers
 #include <TH1F.h>
 
@@ -33,26 +28,6 @@
 #include "DQMServices/Core/interface/DQMStore.h"
 #include "DQMServices/Core/interface/DQMEDAnalyzer.h"
 #include "DQMServices/Core/interface/MonitorElement.h"
-
-// helper functions
-template <typename T>
-static
-const T & get(const edm::Event & event, const edm::EDGetTokenT<T> & token) {
-  edm::Handle<T> handle;
-  event.getByToken(token, handle);
-  if (not handle.isValid())
-    throw * handle.whyFailed();
-  return * handle.product();
-}
-
-template <typename R, typename T>
-static
-const T & get(const edm::EventSetup & setup) {
-  edm::ESHandle<T> handle;
-  setup.get<R>().get(handle);
-  return * handle.product();
-}
-
 
 class TriggerBxVsOrbitMonitor : public DQMEDAnalyzer {
 public:
@@ -202,7 +177,6 @@ void TriggerBxVsOrbitMonitor::analyze(edm::Event const & event, edm::EventSetup 
   int iLS = ls-m_minLS;
   if (iLS >= 0 && iLS < int(m_orbit_bx_all_byLS.size()))
     m_orbit_bx_all_byLS.at(iLS)->Fill(bx,orbit_in_ls);
- 
 
   // monitor the bx distribution for the TCDS trigger types
   size_t size = sizeof(s_tcds_trigger_types) / sizeof(const char *);

--- a/DQM/HLTEvF/plugins/TriggerRatesMonitor.cc
+++ b/DQM/HLTEvF/plugins/TriggerRatesMonitor.cc
@@ -38,26 +38,6 @@
 #include "DQMServices/Core/interface/DQMEDAnalyzer.h"
 #include "DQMServices/Core/interface/MonitorElement.h"
 
-// helper functions
-template <typename T>
-static
-const T & get(const edm::Event & event, const edm::EDGetTokenT<T> & token) {
-  edm::Handle<T> handle;
-  event.getByToken(token, handle);
-  if (not handle.isValid())
-    throw * handle.whyFailed();
-  return * handle.product();
-}
-
-template <typename R, typename T>
-static
-const T & get(const edm::EventSetup & setup) {
-  edm::ESHandle<T> handle;
-  setup.get<R>().get(handle);
-  return * handle.product();
-}
-
-
 class TriggerRatesMonitor : public DQMEDAnalyzer {
 public:
   explicit TriggerRatesMonitor(edm::ParameterSet const &);
@@ -197,7 +177,7 @@ void TriggerRatesMonitor::dqmBeginRun(edm::Run const & run, edm::EventSetup cons
   m_tcds_counts.resize(sizeof(s_tcds_trigger_types)/sizeof(const char *), nullptr);
 
   // cache the L1 trigger menu
-  m_l1tMenu = & get<L1TUtmTriggerMenuRcd, L1TUtmTriggerMenu>(setup);
+  m_l1tMenu = & edm::get<L1TUtmTriggerMenu, L1TUtmTriggerMenuRcd>(setup);
   if (m_l1tMenu) {
     m_l1t_counts.clear();
     m_l1t_counts.resize(GlobalAlgBlk::maxPhysicsTriggers, nullptr);
@@ -339,7 +319,7 @@ void TriggerRatesMonitor::analyze(edm::Event const & event, edm::EventSetup cons
 
   // monitor the L1 triggers rates
   if (m_l1tMenu) {
-    auto const & bxvector = get<GlobalAlgBlkBxCollection>(event, m_l1t_results);
+    auto const & bxvector = edm::get(event, m_l1t_results);
     if (not bxvector.isEmpty(0)) {
       auto const & results = bxvector.at(0, 0);
       for (unsigned int i = 0; i < GlobalAlgBlk::maxPhysicsTriggers; ++i)
@@ -351,7 +331,7 @@ void TriggerRatesMonitor::analyze(edm::Event const & event, edm::EventSetup cons
 
   // monitor the HLT triggers and datsets rates
   if (m_hltConfig.inited()) {
-    edm::TriggerResults const & hltResults = get<edm::TriggerResults>(event, m_hlt_results);
+    edm::TriggerResults const & hltResults = edm::get(event, m_hlt_results);
     if (hltResults.size() == m_hltIndices.size()) {
     } else {
       edm::LogWarning("TriggerRatesMonitor") << "This should never happen: the number of HLT paths has changed since the beginning of the run";

--- a/HLTrigger/HLTcore/src/TriggerExpressionData.cc
+++ b/HLTrigger/HLTcore/src/TriggerExpressionData.cc
@@ -15,28 +15,6 @@
 
 namespace triggerExpression {
 
-template <typename T>
-static 
-const T * get(const edm::Event & event, const edm::EDGetTokenT<T> & token) {
-  edm::Handle<T> handle;
-  event.getByToken(token, handle);
-  if (not handle.isValid()) {
-    auto const & error = handle.whyFailed();
-    edm::LogWarning(error->category()) << error->what();
-    return nullptr;
-  } else {
-    return handle.product();
-  }
-}
-
-template <typename R, typename T>
-static
-const T * get(const edm::EventSetup & setup) {
-  edm::ESHandle<T> handle;
-  setup.get<R>().get(handle);
-  return handle.product();
-}
-
 bool Data::setEvent(const edm::Event & event, const edm::EventSetup & setup) {
 
   // cache the event number
@@ -45,22 +23,22 @@ bool Data::setEvent(const edm::Event & event, const edm::EventSetup & setup) {
   // access L1 objects only if L1 is used
   if (hasL1T()) {
     // cache the L1 GT results objects
-    auto l1t = get<GlobalAlgBlkBxCollection>(event, m_l1tResultsToken);
-    if (not l1t or l1t->size() == 0 or l1t->isEmpty(0)) {
+    auto const& l1t = edm::get(event, m_l1tResultsToken);
+    if (l1t.size() == 0 or l1t.isEmpty(0)) {
       m_l1tResults = nullptr;
       return false;
     }
     if (m_l1tIgnoreMaskAndPrescale)
-      m_l1tResults = & l1t->at(0, 0).getAlgoDecisionInitial();
+      m_l1tResults = & l1t.at(0, 0).getAlgoDecisionInitial();
     else
-      m_l1tResults = & l1t->at(0, 0).getAlgoDecisionFinal();
+      m_l1tResults = & l1t.at(0, 0).getAlgoDecisionFinal();
 
     // cache the L1 trigger menu
     unsigned long long l1tCacheID = setup.get<L1TUtmTriggerMenuRcd>().cacheIdentifier();
     if (m_l1tCacheID == l1tCacheID) {
       m_l1tUpdated = false;
     } else {
-      m_l1tMenu = get<L1TUtmTriggerMenuRcd, L1TUtmTriggerMenu>(setup);
+      m_l1tMenu = & edm::get<L1TUtmTriggerMenu, L1TUtmTriggerMenuRcd>(setup);
       m_l1tCacheID = l1tCacheID;
       m_l1tUpdated = true;
     }
@@ -69,7 +47,7 @@ bool Data::setEvent(const edm::Event & event, const edm::EventSetup & setup) {
   // access HLT objects only if HLT is used
   if (hasHLT()) {
     // cache the HLT TriggerResults
-    m_hltResults = get<edm::TriggerResults>(event, m_hltResultsToken);
+    m_hltResults = & edm::get(event, m_hltResultsToken);
     if (not m_hltResults)
       return false;
 

